### ISSUE TITLE
nyc doesn't forget about deleted/renamed files

### ISSUE THREAD

**Issue by cionnescu** [istanbuljs/nyc#1153]
I'm using nyc to measure the test coverage for a TypeScript project. When I delete/rename a file, nyc will still see the deleted file or the name before renaming. It will consider that the deleted file has 0% coverage. I've tried to delete everything related to istanbul, to restart the computer and re-install everything again, re-build the project and sometimes the problem gets solved, sometimes not. It is not predictive at all.

I expected for nyc to forget about the deleted/renamed files.

Does anyone also faced this problem?

**Comment by bcoe** [comment]
One possible culprit might be that the old files are sitting in a `build/` folder, as I don't think running `tsc compile` gets rid of the prior versions. Try removing your build folder and recompiling when this happens.

**Comment by cionnescu** [comment]
> One possible culprit might be that the old files are sitting in a `build/` folder, as I don't think running `tsc compile` gets rid of the prior versions. Try removing your build folder and recompiling when this happens.

I've tried and it seems to be solved. I also had some problems when changing the code, but I think that deleting 'build/' folder will also solve these problems. It would be great! Thank you!

**Comment by bcoe** [comment]
@cionnescu it might be worth bringing this issue up with TypeScript folks, or I wonder if there's a setting for this already?

tldr; probably `npm run compile` should `rimraf ./build && tsc .`, rather than just running the compiler; but I wouldn't be shocked if there's a TypeScript setting for this.

Feel free to loop me into a conversation on TypeScript, if it turns out this feature doesn't exist.
